website/content/en/docs/advanced-topics/heterogeneous.md

```diff
@@ -0,0 +1,400 @@
+- `testdata/ansible/memcached-operator/config/testing/debug_logs_patch.yaml`
+- `testdata/ansible/memcached-operator/config/testing/manager_image.yaml`
+- `testdata/ansible/memcached-operator/config/testing/pull_policy/Always.yaml`
+- `testdata/ansible/memcached-operator/config/testing/pull_policy/IfNotPresent.yaml`
+- `testdata/ansible/memcached-operator/config/testing/pull_policy/Never.yaml`
+
+ Ansible operators may also create Pod-defining objects dynamically. An example of this kind of object would be in `testdata/ansible/memcached-operator/roles/memcached/tasks/main.yml`.
+ ```yaml
+# tasks file for Memcached
+- name: start memcached
+  kubernetes.core.k8s:
+    definition:
+      ...
+      spec:
+        ...
+        template:
+          ...
+          spec:
+            containers:
+            - name: memcached
+              securityContext:
+                allowPrivilegeEscalation: false
+                capabilities:
+                  drop:
+                    - "ALL"
+              command:
+              - memcached
+              - -m=64
+              - -o
+              - modern
+              - -v
+              image: "docker.io/memcached:1.4.36-alpine"
+              ports:
+                - containerPort: 11211
+              readinessProbe:
+                tcpSocket:
+                  port: 11211
+                initialDelaySeconds: 3
+                periodSeconds: 3
+ ```
+
+ Because the is an embedded yaml, this can be updated by inserting an `affinity` block directly into the yaml like so.
+ ```yaml
+# tasks file for Memcached
+- name: start memcached
+  kubernetes.core.k8s:
+    definition:
+      ...
+      spec:
+        ...
+        template:
+          ...
+          spec:
+            affinity:
+              nodeAffinity:
+                requiredDuringSchedulingIgnoredDuringExecution:
+                  nodeSelectorTerms:
+                  - matchExpressions:
+                    - key: kubernetes.io/arch
+                      operator: In
+                      values:
+                      - amd64
+                      - arm64
+                      - ppc64le
+                      - s390x
+            containers:
+              ...
+ ```
+
+#### Go Operators
+In the `testdata/go/v3` samples app, you'll find several configuration objects that need a nodeAffinity specification in the `config` and `bundle/manifests` directories. The full list is as follows:
+- `testdata/go/v3/memcached-operator/bundle/manifests/memcached-operator.clusterserviceversion.yaml`
+- `testdata/go/v3/memcached-operator/config/default/manager_config_patch.yaml`
+- `testdata/go/v3/memcached-operator/config/default/manager_webhook_patch.yaml`
+- `testdata/go/v3/memcached-operator/config/default/manager_auth_proxy_patch.yaml`
+- `testdata/go/v3/memcached-operator/config/manager/manager.yaml`
```

Comment:
All languages (with the exception of JAVA) scaffold the config directory.
I do not think that we should say in the testdata/ ... we can describe what authors should to do in the default scaffold of their project. We can link the testdata as an example. 

Also, not all these files should be updated to receive the nodeAffinity. 
Only `config/default/manager_config_patch.yaml` and `config/manager/manager.yaml`
The other ones are only patches. 

I think would be nice if you check the PR for update Kubebuilder: https://github.com/kubernetes-sigs/kubebuilder/pull/2906. Those changes will endup here as well, also look at the doc proposed there. I think we can shape this part here.

c/c @jmrodri